transport/tchannel/handler_test.go

```diff
@@ -39,40 +39,92 @@ import (
 	"go.uber.org/yarpc/internal/testtime"
 	pkgerrors "go.uber.org/yarpc/pkg/errors"
 	"go.uber.org/yarpc/yarpcerrors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
 )
 
 func TestHandlerErrors(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
 
 	tests := []struct {
-		format  tchannel.Format
-		headers []byte
-
-		wantHeaders map[string]string
+		desc           string
+		format         tchannel.Format
+		headers        []byte
+		wantHeaders    map[string]string
+		faultyFuncs    map[string]bool
+		wantNumLogs    int
```

Comment:
@twilly I also use this field to count that the expected number of logs are written into the logger. Were you suggesting that I should turn this field into a boolean?

Reply:
Not quite. For example, `wantLogMessage` is a string. If it's not initialized, then it'll be the zero value, which the empty string. You can leverage this behavior by testing like this:

```go
if tt.wantLogMessage != "" {
  assert.Equals(t, logs.FilterMessage(tt.wantLogMessage).Len(), 1)
}
```
If you have multiple messages, you can check with a slice of message strings or LoggedEntry, followed by iterating and comparing.

There's a little test philosophy going on here too. When testing log emissions I generally don't like being too strict, like expecting an exact amount and order of messages. Such a test makes mutating code difficult when we're usually expecting an approximate and fuzzy existence. Usually this is because logs are a loose and flexible interface rather than a rigidly necessary one for a functioning system.